### ISSUE TITLE
jsonProtocol option: Enum Fields not returned

### ISSUE THREAD

**Issue by gmpetrov** [prisma/prisma#18536]
Hello, I just tried the jsonProtocol option and I noticed that Enum fields are not returned unless select {} is used

```js
 // Works
  const datastores = await prisma.datastore.findMany({
    where: {
      ownerId: session?.user?.id,
    },
    select: {
      // Enum Fields
      visibility: true,
      type: true,

      id: true,
      config: true,
      ownerId: true,
      updatedAt: true,
      description: true,
      name: true,
    },,
  });

// No enum fields returned
  const datastores = await prisma.datastore.findMany({
    where: {
      ownerId: session?.user?.id,
    }
  });


// Schema
model Datastore {
  id String @id @default(cuid())

  name        String
  description String
  type        DatastoreType
  datasources AppDatasource[]
  visibility  DatastoreVisibility @default(private)
  apiKeys     DatastoreApiKey[]

  config Json?

  ownerId String? @map("owner_id")
  owner   User?   @relation(fields: [ownerId], references: [id])

  createdAt DateTime @default(now()) @map("created_at")
  updatedAt DateTime @default(now()) @updatedAt @map("updated_at")

  @@map("data_stores")
}

enum DatasourceType {
  web_page
  text
}

enum DatastoreVisibility {
  public
  private
}

```

**Comment by SevInf** [comment]
Hey @gmpetrov. This issues have been fixed in 4.12 release, along with a couple of other improvements to JSON protocol. I'd advice to update and open another issue if you find any problems on new release.

**Comment by gmpetrov** [comment]
@SevInf Good news! I'll give it a try ASAP
